src/schema/v2/__tests__/partner.test.js

```diff
@@ -218,108 +239,236 @@ describe("Partner type", () => {
+                }
               }
             }
           }
-        }
-      `
+        `
 
-      const data = await runQuery(query, context)
+        const data = await runQuery(query, context)
 
-      expect(data).toEqual({
-        partner: {
-          artworksConnection: {
-            pageInfo: {
-              hasNextPage: true,
+        expect(data).toEqual({
+          partner: {
+            artworksConnection: {
+              pageInfo: {
+                hasNextPage: true,
+              },
             },
           },
-        },
+        })
       })
-    })
 
-    it("returns hasNextPage=false when first is above total", async () => {
-      const query = `
-        {
-          partner(id:"bau-xi-gallery") {
-            artworksConnection(first:3) {
-              pageInfo {
-                hasNextPage
+      it("returns hasNextPage=false when first is above total", async () => {
+        const query = gql`
+          {
+            partner(id: "bau-xi-gallery") {
+              artworksConnection(first: 3) {
+                pageInfo {
+                  hasNextPage
+                }
               }
             }
           }
-        }
-      `
+        `
 
-      const data = await runQuery(query, context)
+        const data = await runQuery(query, context)
 
-      expect(data).toEqual({
-        partner: {
-          artworksConnection: {
-            pageInfo: {
-              hasNextPage: false,
+        expect(data).toEqual({
+          partner: {
+            artworksConnection: {
+              pageInfo: {
+                hasNextPage: false,
+              },
             },
           },
-        },
+        })
+      })
+
+      describe("when shallow is false", () => {
+        it("does not return deep artwork data", async () => {
+          const query = gql`
+            {
+              partner(id: "bau-xi-gallery") {
+                artworksConnection(first: 3, shallow: false) {
+                  edges {
+                    node {
+                      slug
+                    }
+                  }
+                }
+              }
+            }
+          `
+          const data = await runQuery(query, context)
+
+          expect(data).toEqual({
+            partner: {
+              artworksConnection: {
+                edges: [
+                  {
+                    node: {
+                      slug: "cara-barer-iceberg",
+                    },
+                  },
+                  {
+                    node: {
+                      slug: "david-leventi-rezzonico",
+                    },
+                  },
+                  {
+                    node: {
+                      slug: "virginia-mak-hidden-nature-08",
+                    },
+                  },
+                ],
+              },
+            },
+          })
+        })
+      })
+      describe("when shallow is true", () => {
+        it("does not return deep artwork data", async () => {
```

Comment:
same as above re: the copy in this block and testing that `partnerArtworksAllLoader` is not called. here, it won't be anyway because `shallow` is true.